crates/ruma-common/src/events/room.rs

```diff
@@ -92,6 +123,45 @@ impl ImageInfo {
     pub fn new() -> Self {
         Self::default()
     }
+
+    /// Create an `ImageInfo` from the given file info, image info and thumbnail.
+    #[cfg(feature = "unstable-msc3552")]
+    pub fn from_extensible_content(
+        file_info: Option<&FileContentInfo>,
+        image: &ImageContent,
+        thumbnail: &Thumbnails,
+    ) -> Option<Self> {
+        if file_info.is_none() && image.is_empty() && thumbnail.is_empty() {
+            None
+        } else {
+            let (mimetype, size) = file_info
+                .map(|info| (info.mimetype.to_owned(), info.size.to_owned()))
+                .unwrap_or_default();
+            let ImageContent { height, width } = image.to_owned();
+            let (thumbnail_src, thumbnail_info) = if thumbnail.is_empty() {
+                (None, None)
+            } else {
+                let thumbnail = &thumbnail.0[0];
```

Comment:
This looks like it should be an `if let Some(thumbnail) = thumbnail.first()` instead. Why do we have that `Thumbnails` wrapper type, and should we potentially remove its `thumbnails` and `is_empty` methods in favor of a `Deref` implementation (if we need to keep the type at all)? Also looks like it should use a `Vec` in its constructor instead of a slice, so we avoid unnecessary copying.

Reply:
We probably don't need the `Thumbnails` and `Captions` types, no. I believe my idea to introduce them was to have one extensible content `m.*` field == one Rust type. But there's not really another justification for them.